Give git merge a hint which directory to merge into since we know exa…

ericpollmann opened the pull request:
…ctly

which one it should use.  This prevents confusing wrong guesses, see:

http://stackoverflow.com/questions/1306595/git-confused-when-merging-an-update-into-my-subtree

ericpollmann committed:
Give git merge a hint which directory to merge into since we know exactly
which one it should use.  This prevents confusing wrong guesses, see:

http://stackoverflow.com/questions/1306595/git-confused-when-merging-an-update-into-my-subtree

csemrau commented:
Giving the `subtree` option for the `subtree` strategy fixed the problem for me. According to the git documentation, `subtree` is a special `recursive` strategy, so it should take the same options. E.g. it also takes `theirs` as option, which is only documented for `recursive`.

ericpollmann commented:
Thank you csemrau!  The vast majority of my strees are dependencies for projects that I generally haven't been changing much, so -Xtheirs is very helpful.

ericpollmann committed:
Update git-stree pull to take extra arguments.
  e.g. git stree pull git-stree -Xtheirs
Update git-stree pull to exit with error message on failure of git merge
  instead of silently exiting with no error message.

(cherry picked from commit e1bfcfc72ebf375b9beca1f51d8ebff9f6bbb6fa)

ericpollmann commented:
...a change to allow "git stree pull [stree_name] -Xtheirs" and show a more informative error message that might suggest its use.

ml31415 commented:
So it's three people now, running into the same bug, and three people, that had success using the same bugfix. I'd really suggest to merge one of the pull requests now. It seems like a simple and straight forward bugfix to me. If there would be a test suite, of course there should also be a test for this. But as there isn't, I'd vote for adding this and the problem is solved.

stela5 commented:
Ref: https://github.com/tdd/git-stree/pull/23#issuecomment-120169894

tdd commented:
Hey all, as stated elsewhere, `-X theirs` isn't one-size-fits-all, as it erases local work inside the subtree. Can't have that. This is needed right now mostly because `master` code didn't prep history merging well enough. The rewrite will fix this, hopefully.
